src/test/ui/or-patterns/feature-gate-or_patterns.stderr

```diff
@@ -0,0 +1,192 @@
+error[E0658]: or-patterns syntax is experimental
+  --> $DIR/feature-gate-or_patterns.rs:5:14
+   |
+LL |         Some(0 | 1 | 2) => {}
+   |              ^^^^^^^^^
+   |
+   = note: for more information, see https://github.com/rust-lang/rust/issues/54883
+   = help: add `#![feature(or_patterns)]` to the crate attributes to enable
+
+error[E0658]: or-patterns syntax is experimental
+  --> $DIR/feature-gate-or_patterns.rs:28:9
+   |
+LL |     let | A | B;
+   |         ^
+   |
+   = note: for more information, see https://github.com/rust-lang/rust/issues/54883
+   = help: add `#![feature(or_patterns)]` to the crate attributes to enable
```

Comment:
* scooby doo head tilt *

Why do we treat this as a separate or pattern instead of just swallowing it in the parser? I would imagine that the intuition of anyone writing this is that the entire thing between `let` and `;` _is_ the pattern. Having two errors here seems silly. Applies to the cases below too.

Reply:
We do swallow it in the parser here: https://github.com/rust-lang/rust/pull/63693/files#diff-af8cd0b0e77c97f09f7a6b59716dcc9cR41 but we also do feature gating to avoid leaking the new syntax into stable. That is, we are not really gating a pattern here, but a *syntactic change*.

Reply:
Could we emit it _only_ if other errors haven't been caught? I'd be even ok with `delay_span_bug` to be honest.